service/history/workflow/mutable_state_impl.go

```diff
@@ -578,6 +578,7 @@ func (e *MutableStateImpl) GetActivityScheduledEvent(
 		e.executionState.RunId,
 		ai.ScheduledEventBatchId,
 		ai.ScheduleId,
+		ai.Version,
```

Comment:
you need to get the version like below:
```bash
version, err := use versionhistory.GetVersionHistoryEventVersion(ai.ScheduleId)
if err != nil {
    return err
}
```

the reason is activity has retry policy, meaning version can change within activity info
although child workflow & request cancel external & signal external has no retry policy, but we still should use the same mechanism in case later version can chance (child workflow & request cancel external & signal external)